application/models/Survey.php

```diff
@@ -247,6 +247,19 @@ public function rules()
     */
     public function fixSurveyAttribute($event)
     {
+        $event = new PluginEvent('afterFindSurvey');
+        $event->set('surveyid',$this->sid) ;
+        App()->getPluginManager()->dispatchEvent($event);
+        // set the attributes we allow to be fixed
+        $allowedAttributes = array( 'template','usecookie', 'allowprev',
+            'showxquestions', 'shownoanswer', 'showprogress', 'questionindex',
+            'usecaptcha', 'showgroupinfo', 'showqnumcode', 'navigationdelay');
+        foreach ($allowedAttributes as $attribute){
+            if (!is_null($event->get($attribute)))
```

Comment:
Nit picky, but `{` should be on same line as `if`. Also, space between `)` and `{` above.